### PR TITLE
Hotelator: Full site 

### PR DESCRIPTION
**Pull request by angeluray**
## Project progress check ✅ 

- Project team member roles established for each member, and divided tasks in a fair way.
- Followed communication skills, and multiple meetings throughout the week for checking overall progress.
- Created ERD.
- Created a Rails API-only project, with its controllers, models, and routes.
- Created API documentation in a human-readable way.
- Deployed API successfully.
- Integrated API with React project.
- Added the Logo of the application(this works as a back button as well)
- Added registration feature.
- Added the user logs in to the website feature using email and password.
- Added navigation panel, so the user can see links to:
  - Hotels that we selected as a theme.
  - "Reserve" form.
  - "My reservations".
  - "Add hotel" (the link is visible to everybody).
  - "Delete hotel" (the link is visible to everybody).
 - Added a main page component that displays all hotels available.
 - Added Detail and Detail form component for allowing users to click a Hotel, see the description, and reserve from that page. (If the user makes a reservation from details, the city and the hotel are already pre-filled in the form)
 - Added the My reservations component to show all the reservations made by a user.
 - Added Reserve component to reserve a Hotel, where based on the city/country that you select you will be able to see all the hotels available.
 - Added Add and Delete hotel component, when the user clicks the "Delete hotel" link in the navigation panel they can see a list of all items with title and "Delete" button, and when the user clicks "Add hotel" a new form is displayed with the user input prefilled. Users can create a new one by adding city, hotel, room type, and date. 
 - Added linters and made sure that everything woks correctly.
 - Changes made doing remote work, asynchronously.
 - Created README file.

**_Note: This project would still be a work in progress for us, since being a developer means improving our current piece of work and making it better, and more efficient. All your feedback will be appreciated 💯, thank you for taking the time to review the project._**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      resources :cities do
+        resources :hotels
+      end
+      resources :room_types, only: [:index]
+
+      resources :hotels
+      resources :cities, only: [:index]
+    end
+  end
+  resources :users, param: :_username
```
Comment (vichuge) [inline config/routes.rb:17]: - It looks like some of the resource commands don't have an only statement, if you forgot it, please add it, if this has a reason, please ignore this. 🐒

**Review by vichuge — COMMENTED** [review]
Hi dear student, 

Good job so far!
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!

mario_ladders


## Suggested changes
_Check the comments under the review._

- I kindly invite to add tests on your project 😄 
image

### [Here](https://github.com/angeluray/hotel-booking-frontend/pull/15) you could see the review for Front-end.

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Kumilachew-g** [comment]
👋 @vichuge, We have done based on your review for the tests

image

File: config/routes.rb (hunk 1 above)
Comment (Kumilachew-g) [reply config/routes.rb:17]: 👋 @vichuge, Thanks for your suggestion. But no problem it works

**Review by Meltrust — APPROVED** [review]
_note to student:  This review outcome only applies to the backend part of the project.  Kindly, go to the frontend PR to view the review comments regarding the frontend app._

#### Hi @angeluray,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Tests are passing✔️ 
- The app works well✔️ 
- Routes have been optimized✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**